Xamarin.Forms.Platform.Android/AppCompat/NavigationPageRenderer.cs

```diff
@@ -612,7 +613,8 @@ Task<bool> SwitchContentAsync(Page view, bool animated, bool removed = false, bo
 					{
 						Fragment currentToRemove = fragments.Last();
 						fragments.RemoveAt(fragments.Count - 1);
-						transaction.Remove(currentToRemove);
+						transaction.Hide(currentToRemove);
```

Comment:
I don't think it does. `SwitchContentAsync` has an `animated` parameter. We set up transitions only if this flag is true:

```
if (animated)
     SetupPageTransition(transaction, !removed);
```

I tested the sample code I provided with non-animated push and pop. It works fine.